datastore/kubernetes.go

```diff
@@ -624,13 +625,23 @@ func (s *DataStore) UpdateConfigMap(configMap *corev1.ConfigMap) (*corev1.Config
 // GetConfigMapRO gets ConfigMap with the given name in s.namespace
 // This function returns direct reference to the internal cache object and should not be mutated.
 // Consider using this function when you can guarantee read only access and don't want the overhead of deep copies
-func (s *DataStore) GetConfigMapRO(namespace, name string) (*corev1.ConfigMap, error) {
-	return s.cfmLister.ConfigMaps(namespace).Get(name)
+func (s *DataStore) GetConfigMapRO(namespace, name string, fromLister bool) (*corev1.ConfigMap, error) {
+	if fromLister {
+		return s.cfmLister.ConfigMaps(namespace).Get(name)
+	}
+	return s.kubeClient.CoreV1().ConfigMaps(s.namespace).Get(context.TODO(), name, metav1.GetOptions{})
 }
 
 // GetConfigMap return a new ConfigMap object for the given namespace and name
```

Comment:
NIT: mentioning why we introduce flag `fromLister`

Reply:
Added. Thank you.